tensorflow/python/distribute/input_lib_type_spec_test.py

```diff
@@ -459,8 +459,27 @@ def f(v):
               strategy_combinations.mirrored_strategy_with_cpu_1_and_2,
               strategy_combinations.mirrored_strategy_with_gpu_and_cpu,
           ],
-          enable_get_next_as_optional=[True, False]))
-  def testTypeSpecForPerReplicaOptions(self, distribution, enable_get_next_as_optional):
+          enable_get_next_as_optional=[True, False],
+          input_options=[
```

Comment:
nit: it might be less verbose to do this instead:
```
combinations.combine(
          mode=["eager"],
          tf_api_version=2,
          distribution=[
              strategy_combinations.mirrored_strategy_with_two_gpus,
              strategy_combinations.mirrored_strategy_with_cpu_1_and_2,
              strategy_combinations.mirrored_strategy_with_gpu_and_cpu,
          ],
          enable_get_next_as_optional=[True, False],
          experimental_place_dataset_on_device=[True,False],
          experimental_prefetch_to_device=[True, False]
)
```
Then you can construct the InputOptions inside the test. This is less verbose and also leads to better test names actually. 
(True, True) combinations will throw error though, so you will have to handle that.